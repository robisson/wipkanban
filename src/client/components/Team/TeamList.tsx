import * as React from "react";
import Typography from "@material-ui/core/Typography";
import Tooltip from "@material-ui/core/Tooltip";
import AddIcon from "@material-ui/icons/AddCircle";
import IconButton from "@material-ui/core/IconButton";
import GroupIcon from "@material-ui/icons/Group";

const TeamList = () => {
  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <GroupIcon /> My teams
        </Typography>
      </div>
      <div style={{ textAlign: "center" }}>
        <Typography component="p" gutterBottom>
          Create your first team
        </Typography>
        <Tooltip title="Add team">
          <IconButton color="default" aria-label="Add team">
            <AddIcon />
          </IconButton>
        </Tooltip>
      </div>
    </div>
  );
};

export default TeamList;
