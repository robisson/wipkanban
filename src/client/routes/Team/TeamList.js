// @flow
import * as React from "react";
import Typography from '@material-ui/core/Typography';
import Tooltip from '@material-ui/core/Tooltip';
import AddIcon from '@material-ui/icons/AddCircle';
import IconButton from '@material-ui/core/IconButton';
import GroupIcon from '@material-ui/icons/Group';

const TeamList = () => {
  return (
    <div>
      <div>
        <Typography color="secondary" variant="title" gutterBottom>
          <GroupIcon/>{' '}
          My teams
        </Typography>
      </div>
      <center>
        <Typography component="p" gutterBottom>
          Create your first team
        </Typography>
        <Tooltip title="Add team">
          <IconButton color="default" aria-label="Add team">
            <AddIcon/>
          </IconButton>
        </Tooltip>
      </center>
    </div>
  )
};

export default TeamList;
