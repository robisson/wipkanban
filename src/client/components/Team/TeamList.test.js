import React from "react";
import TeamList from "./TeamList";
import { createShallow } from "@material-ui/core/test-utils";
import Typography from "@material-ui/core/Typography";
import Tooltip from "@material-ui/core/Tooltip";
import toJson from "enzyme-to-json";

// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("<TeamList /> Component", () => {
  let shallow;
  let wrapper;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<TeamList />);
  });

  it("should render an component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render a div", () => {
    expect(wrapper.type()).toEqual("div");
  });

  it("should render 3 Typogragh and 1 Tooltip", () => {
    expect(wrapper.find(Typography).length).toEqual(2);
    expect(wrapper.find(Tooltip).length).toEqual(1);  });
});
