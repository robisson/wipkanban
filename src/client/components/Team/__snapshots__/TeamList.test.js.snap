// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamList /> Component should render an component correctly 1`] = `
<div>
  <div>
    <WithStyles(Typography)
      color="secondary"
      gutterBottom={true}
      variant="h6"
    >
      <pure(GroupIcon) />
       
      My teams
    </WithStyles(Typography)>
  </div>
  <center>
    <WithStyles(Typography)
      component="p"
      gutterBottom={true}
    >
      Create your first team
    </WithStyles(Typography)>
    <WithStyles(Tooltip)
      title="Add team"
    >
      <WithStyles(IconButton)
        aria-label="Add team"
        color="default"
      >
        <pure(AddCircleIcon) />
      </WithStyles(IconButton)>
    </WithStyles(Tooltip)>
  </center>
</div>
`;
